core/src/com/unciv/logic/civilization/managers/GreatPersonManager.kt

```diff
@@ -49,10 +58,15 @@ class GreatPersonManager : IsPartOfGameInfoSerialization {
         }
 
         for ((key, value) in greatPersonPointsCounter) {
-            val requiredPoints = getPointsRequiredForGreatPerson()
+            val requiredPoints = getPointsRequiredForGreatPerson(key)
             if (value >= requiredPoints) {
                 greatPersonPointsCounter.add(key, -requiredPoints)
-                pointsForNextGreatPerson *= 2
+                for (unique in civInfo.getEquivalentUnit(key).getMatchingUniques(UniqueType.GreatType, stateForConditionals = StateForConditionals.IgnoreConditionals))
```

Comment:
I disagree with ignoring conditionals. If I want a unit to have different types in different eras, for example, why not? If I want a unit to be a great person from a certain point onwards, that's an interesting idea as well!

Reply:
The number of side effects are somewhat numerous, from setting a person to be a great person to a new category towards the end of a game, leading to having enough great person points to get multiple people back to back. In fact, I was considering adding validation check to disallow multiple instances of the unique as it would require the loop even more than what's currently implemented here in order to keep track of all of the groups that the unit could be currently attached to